functions/index.js

```diff
@@ -72,4 +72,39 @@ app.get("/users/:email", async (req, res) => {
   functions.logger.log(docs[0]);
   return res.status(200).json(docs[0]);
 });
+
+app.get("/users", async (req, res) => {
+  functions.logger.log("GET /users");
+
+  const snapshot = await db.collection("users").get();
+  if (snapshot.empty) {
+    functions.logger.log("No matching documents");
+    return res.status(404).send();
+  }
+
+  const docs = [];
+  snapshot.forEach((doc) => {
+    functions.logger.log("doc, ", doc);
+    functions.logger.log("doc id, ", doc.id);
+    docs.push({ id: doc.id, ...doc.data() });
+  });
+  functions.logger.log(docs);
+  return res.status(200).json(docs);
+});
+
+app.delete("/users/:id", async (req, res) => {
+  functions.logger.log("DELETE /users/:id", req.params.id);
+
+  const userId = req.params.id;
+
+  const targetUserDoc = db.collection("users").doc(userId).delete();
+
+  if (!targetUserDoc) {
+    functions.logger.log("No matching entity");
+    return res.status(404).send({ message: "Not Found" });
+  }
+
+  functions.logger.log(`User id ${userId} was deleted`);
+  return res.status(200).send(`User id ${userId} was deleted`);
```

Comment:
I think it would be a good idea to align the return value with json. Multiple formats can be confusing to the receiver when parsing.